src/Agent/NewRelic/Profiler/ModuleInjector/ModuleInjector.h

```diff
@@ -92,5 +117,53 @@ namespace NewRelic { namespace Profiler { namespace ModuleInjector
             sicily::codegen::ByteCodeGenerator generator(tokenizer);
             return generator.TypeToBytes(type);
         }
+
+        static bool EnsureReferenceToMscorlib(IModule& module)
+        {
+            // If this is the NetStandard assembly, it already has a reference to mscorlib, so no need to do anything
+            // Otherwise if this already has a reference to mscorlib, nothing to do.
+            if (module.GetIsThisTheNetStandardAssembly() || module.GetHasRefMscorlib())
+            {
+                return true;
+            }
+
+            try
+            {
+                LogDebug(L"Attempting to Inject reference to mscorlib into netstandard Moddule  ", module.GetModuleName());
+
+                // if the assembly wasn't in the existing references try to define a new one
+                ASSEMBLYMETADATA amd;
+                ZeroMemory(&amd, sizeof(amd));
+                amd.usMajorVersion = 4;
+                amd.usMinorVersion = 0;
+                amd.usBuildNumber = 0;
+                amd.usRevisionNumber = 0;
+
+                auto metaDataAssemblyEmit = module.GetMetaDataAssemblyEmit();
+                mdAssemblyRef assemblyToken;
+                const BYTE pubToken[] = { 0xB7, 0x7A, 0x5C, 0x56, 0x19, 0x34, 0xE0, 0x89 };
+
```

Comment:
This is probably the most controversial item.  Needs to be a strongly referenced assembly with the token.

Reply:
This will probably be okay so long as mscorlib has been using the same strong name key for .net framework 4.5+.

Reply:
We confirmed that the token from framework 4.5 was the same as the one for framework 4.8, so this appears to be a safe thing to hard-code at this point.